hw1/02-palindrome.js

```diff
@@ -1,3 +1,29 @@
-const elem = document.querySelector('input');
+const input = document.querySelector('input');
+const result = document.getElementById('result');
 
-elem.addEventListener('input', handleInput);
+input.addEventListener('input', handleInput);
+
+function handleInput(e) {
+  const inputNumberStr = e.target.value;
+  if (inputNumberStr.length < 1) {
+    result.textContent = '';
+  } else if (isBadInput(inputNumberStr)) {
+    result.style.color = 'red';
+    result.textContent = 'Invalid Input Error: Only positive numbers allowed.';
+  } else if (isPalindrome(inputNumberStr)) {
+    result.style.color = 'green';
+    result.textContent = 'Yes. This is a palindrome!';
+  } else {
+    result.style.color = 'red';
+    result.textContent = 'No. Try again.';
+  }
+}
+
+function isPalindrome(inputNumberStr) {
+  const reverseNumberStr = inputNumberStr.split('').reverse().join('');
+  return inputNumberStr === reverseNumberStr;
+}
+
+function isBadInput(inputNumberStr) {
```

Comment:
_Style_ - Everything in this file looks really great, and I just wanted to note one issue with functions, according to Airbnb's JavaScript style guide. They recommend using

> named function expressions instead of function declarations
>
(reference: [rule 7.1](https://github.com/airbnb/javascript#functions)).  
Following that rule, you might want to look at the three function declarations and also look into placing their definitions before their references.

Reply:
Great catch on this! I did not think to look at the function part of AirBnb for some reason lol. It's interesting how the fix to an unintended behavior in javascript is to give a function two names, it's hard enough thinking of just one name sometimes! 

Fixed in the updated PR